app/models/risk_condition.rb

```diff
@@ -21,6 +22,7 @@ def self.create_condition_for_course_with_type(course_id, type, params, version)
        (type == 3 && (params[:no_submissions_threshold].nil? || params.length != 1)) ||
        (type == 4 && (params[:grade_threshold].nil? ||
                       params[:count_threshold].nil? || params.length != 2))
```

Comment:
it seems like you have missed out a || - this might not have been caught because this function isn't used?